### PR TITLE
Feat(error_recovey): paren_error_recovery

### PR DESCRIPTION
**Pull request by possible-fqz**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [ ] N
- [x] Y #162 

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

When there are parentheses mismatched, the program will not just panic and exit. It will add close parentheses to make them matched.

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [x] Other

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [ ] N
- [x] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [x] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [ ] Other

#### 6. Release note

Please refer to [Release Notes Language Style Guide](https://kusionstack.io/docs/governance/release-policy/) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by possible-fqz** [comment]
recheck

**Comment by possible-fqz** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by possible-fqz** [comment]
recheck

**Comment by He1pa** [comment]
pls complete PR information：
1. add issue link
2. provide scope/description of this PR

File: kclvm/parser/src/parser/expr.rs [hunk 1]
```diff
@@ -1795,7 +1795,11 @@ impl<'a> Parser<'a> {
             TokenKind::CloseDelim(DelimToken::Paren) => {
                 self.bump();
             }
-            _ => self.sess.struct_token_error(
+            // _ => self.sess.struct_token_error(
+            //     &[token::TokenKind::CloseDelim(token::DelimToken::Paren).into()],
```
Comment (He1pa) [inline kclvm/parser/src/parser/expr.rs:1799]: useless comments

**Comment by Peefy** [comment]
At, `kclvm` folder, run `make fmt` to resolve the CI format check point

File: kclvm/parser/src/session/mod.rs [hunk 2]
```diff
@@ -48,6 +49,18 @@ impl ParseSession {
...
+    /// Struct and report an error based on a token and not abort the compiler process.
+     pub fn struct_token_error_recovery(&self, expected: &[String], got: Token) {
+        let pos: Position = self.source_map.lookup_char_pos(got.span.lo()).into();
+        let err = ParseError::UnexpectedToken {
+            expected: expected.iter().map(|tok| tok.into()).collect(),
+            got: got.into(),
+        };
+
+        self.handler.borrow_mut().add_parse_error(err, pos);
+        // panic!("{}", panic_info.to_json_string())
```
Comment (Peefy) [inline kclvm/parser/src/session/mod.rs:61]: remove un-used code.

**Comment by zong-zhe** [comment]
It might be better to change the name of this PR from 'paren_error_recovery' to 'feat(error_recovey): paren_error_recovery'.

**Comment by possible-fqz** [comment]
When there are parentheses mismatched, the program will not just panic and exit. It will add close parentheses to make them matched.

File: kclvm/parser/src/lexer/mod.rs [hunk 3]
```diff
@@ -561,10 +562,17 @@ impl<'a> Lexer<'a> {
...
         if self.indent_cxt.nesting > 0 {
-            self.sess.struct_span_error(
+            self.sess.struct_span_error_recovery(
                 "Unclosed nesting at the end of the file",
                 self.span(start, self.pos),
-            )
+            );
+
+            for _ in 0..self.indent_cxt.nesting {
+                buf.push(Token::new(
```
Comment (He1pa) [inline kclvm/parser/src/lexer/mod.rs:571]: I'm not sure it's necessary to distinguish between paren, bracket and brace here @Peefy

Comment (Peefy) [reply kclvm/parser/src/lexer/mod.rs:571]: Pls add more test cases about paren, bracket and brace parse.

**Comment by Peefy** [comment]
Close this and reopen with #216

**Pull request closed without merging**